File: 2.Add_Two_Numbers/Add_Two_Numbers.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var addTwoNumbers = function(l1, l2) {
    var dummy = new ListNode(-1);
    var current = dummy;
    var carry = 0;
    while(l1 !== null || l2 !== null || carry !== 0){ //IMPORTANT!!! Different from another languages.
    	var sum = carry;

    	if(l1 !== null){
    		sum += l1.val;
    		l1 = l1.next;
    	}

    	if(l2 !== null){
    		sum += l2.val;
    		l2 = l2.next;
    	}

    	var num = sum%10;
    	carry = parseInt(sum/10); //IMPORTANT!!! Different from another languages.

    	current.next = new ListNode(num);
    	current = current.next;
    }
    current.next = null;
    return dummy.next;
};

File: 2.Add_Two_Numbers/Add_Two_Numbers.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        flag = 0
        dummy = ListNode(-1)
        current = dummy
        while l1 and l2:
            v = l1.val + l2.val + flag
            flag, num = v/10, v%10
            current.next = ListNode(num)
            current = current.next
            l1 = l1.next
            l2 = l2.next
        if l1 and not l2:
            while l1:
                v = l1.val + flag
                flag, num = v/10, v%10
                current.next = ListNode(num)
                current = current.next
                l1 = l1.next
        elif not l1 and l2:
            while l2:
                v = l2.val + flag
                flag, num = v/10, v%10
                current.next = ListNode(num)
                current = current.next
                l2 = l2.next
        if flag == 1:
            current.next = ListNode(1)
            current = current.next
        current.next = None
        return dummy.next